flang/lib/Lower/SymbolMap.h

```diff
@@ -226,25 +236,24 @@ class SymMap {
   }
 
   /// Find `symbol` and return its value if it appears in the current mappings.
-  SymbolBox lookupSymbol(semantics::SymbolRef sym) {
-    auto iter = symbolMap.find(&*sym);
-    if (iter != symbolMap.end())
-      return iter->second;
-    // Follow host association
-    if (const auto *details =
-            sym->detailsIf<Fortran::semantics::HostAssocDetails>())
-      return lookupSymbol(details->symbol());
-    return SymbolBox::None{};
-  }
+  SymbolBox lookupSymbol(semantics::SymbolRef sym);
 
   /// Remove `sym` from the map.
-  void erase(semantics::SymbolRef sym) { symbolMap.erase(&*sym); }
+  /// FIXME: Get rid of this as it's likely not what is expected.
+  void erase(semantics::SymbolRef sym) { symbolMapStack.back().erase(&*sym); }
```

Comment:
At least for the usage of erase for stmt functions, this could now be replaced by a pushScope [before adding the dummies](https://github.com/flang-compiler/f18-llvm-project/blob/82ce99fbf720256c016351225e58291b2e63df74/flang/lib/Lower/ConvertExpr.cpp#L1405) and popScope [after the Expr evaluation](https://github.com/flang-compiler/f18-llvm-project/blob/82ce99fbf720256c016351225e58291b2e63df74/flang/lib/Lower/ConvertExpr.cpp#L1420).